bootloader/tests/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# this CMakeLists.txt file is only used when host-compiling to build tests
+
+find_package(Catch2 REQUIRED)
+include(CTest)
+include(Catch)
+include(AddBuildAndTestTarget)
+
+add_executable(
+        bootloader
+        test_main.cpp
+)
+
+target_include_directories(bootloader PUBLIC ${CMAKE_SOURCE_DIR}/include)
+set_target_properties(bootloader
+        PROPERTIES
+        CXX_STANDARD 20
+        CXX_STANDARD_REQUIRED TRUE)
+
+target_compile_options(bootloader
+        PUBLIC
+        -Wall
+        -Werror
+        -Weffc++
+        -Wreorder
+        -Wsign-promo
+        -Wextra-semi
+        -Wctor-dtor-privacy
+        -fno-rtti)
+target_link_libraries(bootloader Catch2::Catch2)
```

Comment:
should this maybe link common-simulation as well for any future tests?

Reply:
Good point. Maybe if I'll wait until we're sure we need that library.